# Datasets used in FORAlign

## Datasets source

We use [`GMGC`](https://github.com/malabz/GMGC-data), [`HIV`](https://github.com/malabz/HIV-data), [`CoVID-19`](https://www.gisaid.org/) and [`MPoX`](https://github.com/malabz/MPoxVR_dataset) datasets to measure our methods.

Citations:
- GMGC: Coelho, L.P., et al. Towards the biogeography of prokaryotic genes. Nature 601, 252â€“256 (2022).
- HIV: https://www.hiv.lanl.gov/
- CoVID-19: Shu YL, McCauley J. GISAID: Global initiative on sharing all influenza data - from vision to reality. Eurosurveillance. 2017; 22(13):2-4.
- MPoX: Ma Y, Chen M, Bao Y, et al. MPoxVR: A comprehensive genomic resource for monkeypox virus variant surveillance. The Innovation, 2022, 3(5).

## Test cases used in WFA2-test program provided by FORAlign

The generated test data is saved in `test_case.tar.xz`. This data also saved at [`Zenodo`](https://zenodo.org/records/10906782). The following command provides the way to decompress the file:

```bash
mkdir test_case
tar xvf test_case.tar.xz -C ./test_case
```

For every file in `test_case` folder, the name of file is the source of test cases (e.g., `GMGC_20000__GMGC_10000` means the file is generated by test case `GMGC_20000` and `GMGC_10000` for pairwise sequence alignment, the first sequence is randomly selected in `GMGC_20000` file, the second sequence is randomly selected in `GMGC_10000` file), the `*.ans` file is CIGAR string generated by `Benchmark` (i.e., `gap-affine-swg` in WFA2).

## Test programs

Test programs has compiled in C++20, and the program is static linked (i.e. no extra dependencies, can be runned in any Linux platforms). The programs has stored in `wfa2-test-prog` folder.

## Test results

All tests are tested in Linux. The results are generated by `align_benchmark` and `multiswg_benchmark` programs compiled by `wfa2-test` folder in FORAlign.

### Organized results
The organized results is saved in `test-result/json` folder. The following table shows the content for these datas:

|File Name|Content Info|Used in test|
|:-:|:-:|:-:|
|9gpu_*core.json|The test result from 9gpu device use the determined number of cores|all of tests|
|9gpu_benchmark.json|`gap-affine-swg` result generated by `128 cores`|benchmark for anaylzing datas|
|hpc_56core_cpucnt_anaysis.json|The test result which the CPU percent tested by `top` is more than 95% generated by `56 cores`|condition_variable and Speedup Ratio for `swg`|
|hpc_56core_simpletest.json|The test result which the CPU percent tested by `top` is more than 95% generated by `56 cores`, not tested WFA series|t-blocks|
|hpc_80core_cpucnt_anaysis.json|The test result which the CPU percent tested by `top` is more than 95% generated by `80 cores`|all of tests except Speedup Ratio for `swg`|
|hpc_80core_max_memory.json|The maximum memory in `Benchmark` methods|Methods selection|
|singlepc_1_12core.json|The test result generated by `12 cores`|all of tests except Speedup Ratio for `swg`|
|singlepc_2_16core.json|The test result generated by `16 cores`|supplemental data|
|singlepc_2_24core.json|The test result generated by `24 cores`|all of tests except Speedup Ratio for `swg`|

### Raw Result in `test-result/raw.tar.xz` file and compress

The raw test cases is in `test-result/raw.tar.xz`. The following command provides the way to decompress the file:

```bash
tar xvf test-result/raw.tar.xz
```

Files will in `test-result/raw` folder.
